src/compiler/generate-common-tests.xsl

```diff
@@ -639,7 +639,25 @@
 
          <!-- Replace x:like with specified scenario's child elements -->
          <xsl:when test="self::x:like">
-            <xsl:apply-templates select="key('scenarios', x:label(.))/element()" mode="#current" />
+            <xsl:variable name="label" as="element(x:label)" select="x:label(.)" />
+            <xsl:variable name="scenario" as="element(x:scenario)*" select="key('scenarios', $label)" />
+            <xsl:choose>
+               <xsl:when test="empty($scenario)">
+                  <xsl:sequence select="error(xs:QName('x:XSPEC009'),
+                     concat(name(), ': Scenario not found: ', $label))" />
+               </xsl:when>
+               <xsl:when test="$scenario[2]">
+                  <xsl:sequence select="error(xs:QName('x:XSPEC010'),
+                     concat(name(), ': Multiple scenarios found: ', $label))" />
+               </xsl:when>
+               <xsl:when test="$scenario intersect ancestor::x:scenario">
+                  <xsl:sequence select="error(xs:QName('x:XSPEC011'),
+                     concat(name(), ': Scenario is looping: ', $label))" />
```

Comment:
How about saying why it's looping, such as "Reference to ancestor creates infinite loop"?

Reply:
ec998ed0758fac9ea2581d7385fdc73b1cfb3ff0 updated the message to "Reference to ancestor scenario creates infinite loop".